swarm/src/behaviour.rs

```diff
@@ -191,6 +192,35 @@ pub trait NetworkBehaviour: 'static {
         vec![]
     }
 
+    // TODO: Make sure this and all the methods below are really implemented across all behaviours.
+    fn review_pending_connection(
+        &mut self,
+        _peer_id: Option<PeerId>,
+        // TODO: Maybe an iterator is better?
+        _addresses: &[Multiaddr],
+        _endpoint: Endpoint,
```

Comment:
Instead of passing the endpoint, would it make sense to have two different callbacks, one for incoming and one for outgoing?